Fix Hit Tests in RenderEditable when Scrolled

creativecreatorormaybenot opened the pull request:
This fixes `hitTestChildren` in `RenderEditable` by respecting the scroll `offset` (`_paintOffset` for applying the scroll axis).

Before, hit tests were broken when the `EditableText`/`TextField` was scrolled.

### Issues

Fixes #81038.

### Tests

The first test case I added was passing before (making sure that my change does not make that fail) and the latter two test cases were failing before.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

<!-- Links -->
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

LongCatIsLooong commented:
The change looks good to me. Could you create a `RenderEditable` with a non-zero offset, and call `hitTest` on it to see if the method is doing the right thing? It's probably the easiest to follow the example of tests in "test/rendering/editable_test.dart"

creativecreatorormaybenot commented:
@LongCatIsLooong Thanks for the hint. I am working on the tests and will probably have them ready in a few 👍

creativecreatorormaybenot committed:
Add failing test cases

creativecreatorormaybenot committed:
Add effectivePosition

creativecreatorormaybenot commented:
I added the test cases 👍 Note that the first test case I added was passing before (making sure that my change does not make that fail) and the latter two test cases were failing before.

LongCatIsLooong commented on packages/flutter/test/rendering/editable_test.dart:
```diff
@@ -3330,6 +3331,137 @@ void main() {
+        final BoxHitTestResult result = BoxHitTestResult();
+        // At x=5, we should hit "B" as we are scrolled to the left by 5
+        // pixels.
+        editable.hitTest(result, position: const Offset(5.0, 0));
```
5.0 is right on the boundary and can go either way. This may have to do with the web test failure: https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket.appspot.com/8849033682682636208/+/steps/run_test.dart_for_web_tests_shard_and_subshard_3/0/logs/stdout/0. Could you try 6 or something instead?

creativecreatorormaybenot replied:
Sure, I had it at `5.1` first but it felt somehwhat arbitrary. But it does make sense to use a safe value 👍

creativecreatorormaybenot committed:
Update offset
